Movement.cpp

```diff
@@ -9,105 +9,105 @@
-  double encoder1 = 0;
-  double encoder2 = 0;
-  double pwm_left_final = 0;
-  double pwm_right_final = 0;
-  double pwm_left = 0;
-  double pwm_right = 0;
-  double pwm_right_BNO = 0;
-  double pwm_left_BNO = 0;
-  double pwm_right_ultrasonic_right_up = 0;
-  double pwm_left_ultrasonic_right_up = 0;
-  double pwm_left_ultrasonic_right_down = 0;
-  double pwm_right_ultrasonic_right_down = 0;
-  double pwm_left_ultrasonic_left_up = 0;
-  double pwm_right_ultrasonic_left_up = 0;
-  double pwm_left_ultrasonic_left_down = 0;
-  double pwm_right_ultrasonic_left_down = 0;
-  const double current_distance_right_up = sensor.getDistanceRightUp();
-  const double current_distance_right_down = sensor.getDistanceRightDown();
-  const double current_distance_left_up = sensor.getDistanceLeftUp();
-  const double current_distance_left_down = sensor.getDistanceLeftDown();
-  double error_right_up = control.getErrorUltrasonic(current_distance_right_up, desireUltrasonic);
-  double error_right_down = control.getErrorUltrasonic(current_distance_right_down, desireUltrasonic);
-  double error_left_up = control.getErrorUltrasonic(current_distance_left_up, desireUltrasonic);
-  double error_left_down = control.getErrorUltrasonic(current_distance_left_down, desireUltrasonic);
-  double current_angle = control.getAngleBNOX();
-  double errorBNO = control.getAngleError(current_angle, desire);
+  eCount1 = 0;
+  eCount2 = 0;
 
   do {
+    double pwm_left_final = 0;
+    double pwm_right_final = 0;
+    double pwm_right_BNO = 0;
+    double pwm_left_BNO = 0;
+    double pwm_right_ultrasonic_right_up = 0;
+    double pwm_left_ultrasonic_right_up = 0;
+    double pwm_left_ultrasonic_right_down = 0;
+    double pwm_right_ultrasonic_right_down = 0;
+    double pwm_left_ultrasonic_left_up = 0;
+    double pwm_right_ultrasonic_left_up = 0;
+    double pwm_left_ultrasonic_left_down = 0;
+    double pwm_right_ultrasonic_left_down = 0;
+    double current_angle_x = logic.getAngleBNOX();  
+    const double current_distance_right_down = sensor.getDistanceRightDown();
+    const double current_distance_left_up = sensor.getDistanceLeftUp();
+    const double current_distance_left_down = sensor.getDistanceLeftDown();
+    const double current_distance_right_up = sensor.getDistanceRightUp();
+
     if (current_distance_right_up < 20) {
+        double error_right_up = control.getErrorUltrasonic(current_distance_right_up, desireUltrasonic);
         if (error_right_up > 0) {
-        pwm_right = kLimit_inf_pwm;
-        pwm_left_ultrasonic_right_up = kP2 * error_right_up;
+          pwm_right_final = kLimit_inf_pwm;
+          pwm_left_ultrasonic_right_up = kPAdvance * error_right_up;
         }
         else {
-        pwm_left = kLimit_inf_pwm;
-        pwm_right_ultrasonic_right_up = kP2 * error_right_up;
+          pwm_left_final = kLimit_inf_pwm;
+          pwm_right_ultrasonic_right_up = kPAdvance * error_right_up;
         }
```

Comment:
What's the difference between the if and the else?

Reply:
This occurs throughout the whole function.

Reply:
If one is positive, add pwm to the left enginees, and if it's negative, add pwm to the right enginees. Did you meant that?